src/iexamine.cpp

```diff
@@ -2332,6 +2332,97 @@ void iexamine::kiln_full( player &, const tripoint &examp )
     add_msg( _( "It has finished burning, yielding %d charcoal." ), result.charges );
 }
 
+void iexamine::autoclave_empty( player &p, const tripoint &examp )
+{
+    furn_id cur_autoclave_type = g->m.furn( examp );
+    furn_id next_autoclave_type = f_null;
+    if( cur_autoclave_type == furn_id( "f_autoclave" ) ) {
+        next_autoclave_type = furn_id( "f_autoclave_full" );
+    } else {
+        debugmsg( "Examined furniture has action autoclave_empty, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+
+
+    bool water_is_enough = g->u.crafting_inventory().has_charges( "water", 64 );
+    map_stack items = g->m.i_at( examp );
+    static const std::string filthy( "FILTHY" );
+    bool filthy_cbms = std::all_of( items.begin(), items.end(), []( const item & i ) {
+        return i.has_flag( filthy ) && i.is_bionic();
+    } );
+
+    if( items.empty() ) {
+        add_msg( _( "This autoclave is empty..." ) );
+        return;
+    }
+    if( !filthy_cbms ) {
+        add_msg( m_bad,
+                 _( "You need to remove all non-filthy non-CBM items from the autoclave to start the program." ) );
+        return;
+    } else if( !water_is_enough ) {
+        add_msg( m_bad, _( "You need 64 charges of water for the autoclave." ) );
```

Comment:
The number and component in this message should not be hardcoded.  You can use `requirement::list_missing()` here

Reply:
damn I was sure I did that though, must have messed up a commit